File: RelatedWordsAPI/Models/Word.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace RelatedWordsAPI.Models
{
    public class Word : IEquatable<Word>
    {

        public int WordId { get; set; }
        public int ProjectId { get; set; }
        public Project Project { get; set; }
        public string WordContent { get; set; }

        public ICollection<WordSentence> WordSentences { get; }
        public ICollection<WordPage> WordPages { get; }
        public Word() { }
        public Word(string wordContnet, Project project)
        {
            WordContent = wordContnet;
            Project = project;
        }
        public override int GetHashCode()
        {
            return WordContent.GetHashCode() ^ ProjectId.GetHashCode();
        }

        public bool Equals(Word word)
        {
            return word.ProjectId == ProjectId && word.WordContent == WordContent;
        }
    }
}

File: RelatedWordsUI/related-words-ui/src/models/Word.js
import isDefinedOrNull from './helpers/isDefinedOrNull';

class Word {
  constructor(props) {
    this.wordId = isDefinedOrNull(props.wordId);
    this.projectId = isDefinedOrNull(props.projectId);
    this.wordContent = isDefinedOrNull(props.wordContent);
  }
}

export default Word;